packages/edit-site/src/components/global-styles/font-library-modal/context.js

```diff
@@ -101,15 +101,18 @@ function FontLibraryProvider( { children } ) {
 				.sort( ( a, b ) => a.name.localeCompare( b.name ) )
 		: [];
 
-	const customFonts = fontFamilies?.custom
-		? fontFamilies.custom
+	const baseCustomFonts = libraryFonts
+		? libraryFonts
 				.map( ( f ) => setUIValuesNeeded( f, { source: 'custom' } ) )
 				.sort( ( a, b ) => a.name.localeCompare( b.name ) )
 		: [];
 
-	const baseCustomFonts = libraryFonts
-		? libraryFonts
+	const customFonts = fontFamilies?.custom
+		? fontFamilies.custom
 				.map( ( f ) => setUIValuesNeeded( f, { source: 'custom' } ) )
+				.filter( ( f ) =>
+					baseCustomFonts.some( ( bf ) => bf.slug === f.slug )
+				) // Ensures that the font actually exists in the library.
```

Comment:
Do you think this is necessary? I think it could lead to some inconsistencies like not displaying a font that is still activated (in global styles) but not installed.

Reply:
> I think it could lead to some inconsistencies like not displaying a font that is still activated (in global styles) but not installed.

This is what happens now / on trunk. Without this line — if the font uninstallation fails, it will still display the activated font in the typography sidebar.

Reply:
I see what you mean though, I'll remove it for now and find another way to address it if there are issues later.

Reply:
Not quite — the way this is failing, is that the font gets uninstalled successfully on the server side, but not in global styles since that is handled client side and that's where the bug is. The result is the font family is uninstalled and deleted, but is still defined in global styles / shows up in the sidebar.